examples/Mediawiki/ListUsers.yaml

```diff
@@ -0,0 +1,28 @@
+suite: Listing Users
+description: Use API:Allusers to list all users
+type: RPC
+
+use-fixtures:
+  - Alice
+  - Bob
+  - RootUser
+
+tests:
+  - description: Get a list of all registered users
+    interaction:
+      - request:
+          path: api.php
+          parameters:
+            action: query
+            list: allusers
+            format: json
+        response:
+          body:
+            query:
+              allusers:
+                - userid: p/var: Alice/id
+                  name: p/var: Alice/name
+                - userid: p/var: Bob/id
+                  name: p/var: Bob/name
+                - userid: p/var: RootUser/id
+                  name: p/var: Rootuser/name
```

Comment:
This is tricky. We know that these users must exist, but we don't know that no other users exist. There may potentially be hundreds of users before Alice, even, so many that Alice gets pushed out of the first page of results (that reminds me - we have no tests for paging. I guess that's for later).

The only way I can think of is to specifically create users with an unlikely prefix, and then use the auprefix parameter to fetch only them.

For the list of all users, all we could do is check that there are at least three. For that, we'd need a new assertion, something like this:
  allusers: p/min-list-size: 3
That's very specific, but it's good to note that we found a new kind of assertion we might need. A case in which the approach of comparing structures doesn't work well. It seems that lists in general are problematic, it really only works for objects/maps.

Reply:
I've updated it. When you get a chance can you look it over?

> That's very specific, but it's good to note that we found a new kind of assertion we might need. 

I agree this is one assertion we'll need to add but I think we should also expand Phester to allow users to add their own testing functions in some specified file. This way we do our best to keep Phester simple while allowing others to extend it as necessary for their application.

Reply:
> I agree this is one assertion we'll need to add but I think we should also expand Phester to allow users to add their own testing functions in some specified file. This way we do our best to keep Phester simple while allowing others to extend it as necessary for their application.

Yes, this is definitely a useful "extension point" to provide.